### PR TITLE
feat: add global flag for running dep sequentially

### LINKED ISSUE #2380: Deploy with `--max-concurrent-builds` or `--build-sequential` does not apply to concurrent dependency builds
**What happened?**
I have a monorepo with API gateway. The gateway project depends on multiple service projects. Each project, including gateway, has at least one image to build.

When I run either `devspace deploy --build-sequential` or `devspace deploy --max-concurrent-builds 1` , devspace kicks off builds for mulitiple dependency projects at once, resulting in heavy parallelism, high memory usage by docker and recurring build failures.

**What did you expect to happen instead?**
I expected both `devspace deploy --build-sequential` and `devspace deploy --max-concurrent-builds 1` to kick off 1 build at a time. I would like developers to choose concurrency with `devspace deploy --max-concurrent-builds X`,  to control CPU load and memory usage on their machine.

**How can we reproduce the bug?** (as minimally and precisely as possible)
I have prepared a reproduction repo, just follow the steps from the readme.

[https://github.com/gustaff-weldon/devspace-build-concurrency](https://github.com/gustaff-weldon/devspace-build-concurrency)

**Local Environment:**

* DevSpace Version: 6.1.1
* Operating System: mac
* ARCH of the OS: x86_64

**Kubernetes Cluster:**

* local Docker Desktop cluster, but builds happen outside of it

**Anything else we need to know?**

/kind bug

### PR DESCRIPTION
**Pull request by pratikjagrut**
**What issue type does this pull request address?** (keep at least one, remove the others) 
/kind feature

**What does this pull request do? Which issues does it resolve?** (use `resolves #<issue_number>` if possible) 
resolves #2380 

**Please provide a short message that should be published in the DevSpace release notes**
Added global flag `--sequential-dependencies` which deploys dependencies sequentially.

### CODE REVIEW

File: pkg/devspace/pipeline/pipeline.go [hunk 1]
```diff
@@ -224,10 +224,13 @@ func (p *pipeline) StartNewDependencies(ctx devspacecontext.Context, dependencie
 
 	// Start sequentially
 	if options.Sequential {
+		ctx.Log().Info("Deploying dependencies sequentially")
```
Comment (lizardruss) [inline pkg/devspace/pipeline/pipeline.go:227]: Can we switch both log statements to `Debug()`/`Debugf()`?

**Review by lizardruss — CHANGES_REQUESTED** [review]
Very small change requested, but otherwise LGTM!

File: pkg/devspace/pipeline/pipeline.go (hunk 1 above)
Comment (pratikjagrut) [reply pkg/devspace/pipeline/pipeline.go:227]: Okay.

Comment (pratikjagrut) [reply pkg/devspace/pipeline/pipeline.go]: Done

**Review by FabianKramm — CHANGES_REQUESTED** [review]

**Review by FabianKramm — APPROVED** [review]

**Pull request merged**